src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -768,12 +844,13 @@ public void sendAndNotify(WebSocketMessageDTO msg) throws IOException {
 			out = remoteListener.getOutputStream();
 		}
 	
+		notifyMessageSenderListeners(message, initiator);
 		if (message.forward(out)) {
 			notifyMessageObservers(message);
 		}
 	}
 
-	public boolean send(WebSocketMessageDTO msg) throws IOException {
+	public boolean send(WebSocketMessageDTO msg, int initiator) throws IOException {
```

Comment:
I implemented the new method `WebSocketProxy#send(WebSocketMessageDTO msg, Initiator initiator)`.
I also marked `WebSocketProxy#send(WebSocketMessageDTO msg)` as `@Deprecated`.